src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -663,6 +663,109 @@ public int GetLength(MemoryPoolIterator end)
             }
         }
 
+        public static unsafe void FastByteCopy(byte[] srcArray, uint srcOffset, byte[] destArray, uint destOffset, uint byteCount)
+        {
+            Debug.Assert(srcOffset + byteCount <= srcArray.Length);
```

Comment:
Asserts on blocks should be testing `block.Data.Count` rather than `block.Array.Length`